cmd/config-current.go

```diff
@@ -720,6 +725,55 @@ func applyDynamicConfigForSubSys(ctx context.Context, objAPI ObjectLayer, s conf
 	return nil
 }
 
+// autoGenerateRootCredentials generates root credentials deterministically if
+// a KMS is configured, no manual credentials have been specified and if root
+// access is disabled.
+func autoGenerateRootCredentials() {
+	if GlobalKMS == nil {
+		return
+	}
+	if globalAPIConfig.permitRootAccess() || !globalActiveCred.Equal(auth.DefaultCredentials) {
+		return
+	}
```

Comment:
Should we also check if root creds env vars are set like `MINIO_ROOT_USER` and `MINIO_ROOT_PASSWORD`? If they are I don't think we should generate via KMS

Reply:
globalActiveCreds already overrides root creds via env